packages/firestore/src/local/simple_db.ts

```diff
@@ -249,12 +186,91 @@ export class SimpleDb {
     }
   }
 
+  /**
+   * Opens the specified database, creating or upgrading it if necessary.
+   */
+  async ensureDb(): Promise<IDBDatabase> {
+    if (!this.db) {
+      logDebug(LOG_TAG, 'Opening database:', this.name);
+      this.db = await new Promise<IDBDatabase>((resolve, reject) => {
+        // TODO(mikelehen): Investigate browser compatibility.
+        // https://developer.mozilla.org/en-US/docs/Web/API/IndexedDB_API/Using_IndexedDB
+        // suggests IE9 and older WebKit browsers handle upgrade
+        // differently. They expect setVersion, as described here:
+        // https://developer.mozilla.org/en-US/docs/Web/API/IDBVersionChangeRequest/setVersion
+        const request = indexedDB.open(this.name, this.version);
+
+        request.onsuccess = (event: Event) => {
+          const db = (event.target as IDBOpenDBRequest).result;
+          resolve(db);
+        };
+
+        request.onblocked = () => {
+          reject(
+            new IndexedDbTransactionError(
```

Comment:
This is mostly a copy of the code above, but this line changed from FirestoreError to IndexedDbTransactionError.